layers/descriptor_validation.cpp

```diff
@@ -3308,3 +3396,913 @@ bool CoreChecks::VerifyWriteUpdateContents(const DescriptorSet *dest_set, const
+                                 "vkCmdBindDescriptorBuffersEXT(): pBindingInfos[%" PRIu32
+                                 "].usage does not match the buffer from which the address was queried",
+                                 i);
+            }
+        }
+
+        const auto *buffer_handle = LvlFindInChain<VkDescriptorBufferBindingPushDescriptorBufferHandleEXT>(pBindingInfos[i].pNext);
+        if (!phys_dev_ext_props.descriptor_buffer_props.bufferlessPushDescriptors &&
+            (pBindingInfos[i].usage & VK_BUFFER_USAGE_PUSH_DESCRIPTORS_DESCRIPTOR_BUFFER_BIT_EXT) && !buffer_handle) {
+            skip |= LogError(device, "VUID-VkDescriptorBufferBindingInfoEXT-bufferlessPushDescriptors-08056",
+                             "vkCmdBindDescriptorBuffersEXT(): pBindingInfos[%" PRIu32
+                             "].pNext does not contain a VkDescriptorBufferBindingPushDescriptorBufferHandleEXT structure, but "
+                             "VkPhysicalDeviceDescriptorBufferPropertiesEXT::bufferlessPushDescriptors is VK_FALSE and usage "
+                             "contains VK_BUFFER_USAGE_PUSH_DESCRIPTORS_DESCRIPTOR_BUFFER_BIT_EXT",
+                             i);
+        }
+
+        if (SafeModulo(pBindingInfos[i].address, phys_dev_ext_props.descriptor_buffer_props.descriptorBufferOffsetAlignment) != 0) {
+            skip |= LogError(device, "VUID-VkDescriptorBufferBindingInfoEXT-address-08057",
+                             "vkCmdBindDescriptorBuffersEXT(): pBindingInfos[%" PRIu32 "].address (%" PRIuLEAST64 ") is not aligned "
+                             "to VkPhysicalDeviceDescriptorBufferPropertiesEXT::descriptorBufferOffsetAlignment (%" PRIuLEAST64 ")",
+                             i, pBindingInfos[i].address, phys_dev_ext_props.descriptor_buffer_props.descriptorBufferOffsetAlignment);
+        }
+
+        if (buffer_handle && phys_dev_ext_props.descriptor_buffer_props.bufferlessPushDescriptors) {
+            skip |= LogError(device, "VUID-VkDescriptorBufferBindingPushDescriptorBufferHandleEXT-bufferlessPushDescriptors-08059",
+                             "vkCmdBindDescriptorBuffersEXT(): pBindingInfos[%" PRIu32
+                             "].pNext contains a VkDescriptorBufferBindingPushDescriptorBufferHandleEXT structure, "
+                             "but VkPhysicalDeviceDescriptorBufferPropertiesEXT::bufferlessPushDescriptors is VK_TRUE",
+                             i);
+        }
+
+        if (bindingInfo.usage & VK_BUFFER_USAGE_SAMPLER_DESCRIPTOR_BUFFER_BIT_EXT) {
+            ++num_sampler_buffers;
+
+            if (!buffer_state || (buffer_state->createInfo.usage & VK_BUFFER_USAGE_SAMPLER_DESCRIPTOR_BUFFER_BIT_EXT) == 0)
+                skip |= LogError(device, "VUID-VkDescriptorBufferBindingInfoEXT-usage-08122",
+                                 "If usage includes VK_BUFFER_USAGE_SAMPLER_DESCRIPTOR_BUFFER_BIT_EXT, address must be an address "
```

Comment:
For the other ones below as well

```suggestion
                                 "vkCmdBindDescriptorBuffersEXT():  If usage includes VK_BUFFER_USAGE_SAMPLER_DESCRIPTOR_BUFFER_BIT_EXT, address must be an address "
```